stardog/admin.py

```diff
@@ -484,36 +557,114 @@ def new_virtual_graph(self, name, mappings, options=None, datasource=None, db=No
 
         Examples:
             >>> admin.new_virtual_graph(
+                  'users', MappingFile('mappings.sms'), None, 'my_datasource'
+                )
+            >>> admin.new_virtual_graph(
+                  'users', MappingFile('mappings.ttl','SMS2'), None, 'my_datasource'
+                )
+            >>> admin.new_virtual_graph(  #DEPRECATED
                   'users', File('mappings.ttl'),
                   {'jdbc.driver': 'com.mysql.jdbc.Driver'}
                 )
         """
 
         if mappings:
-            with mappings.data() as data:
-                mappings = data.read().decode() if hasattr(data, "read") else data
+            if mappings.syntax:
+                if options:
+                    options["mappings.syntax"] = mappings.syntax
+                else:
+                    options = {"mappings.syntax": mappings.syntax}
 
-        meta = {}
-        meta["name"] = name
-        meta["mappings"] = mappings
-        if options is not None:
+            with mappings.data() as data:
+                if hasattr(data, "read"):
+                    r = data.read()
+                    mappings = r.decode() if hasattr(r, "decode") else r
+                else:
+                    mappings = data
+
+        meta = {"name": name, "mappings": mappings}
+        if options:
```

Comment:
I don't think it changes much here, but it's usually a best practice to compare these to "is not None", see https://peps.python.org/pep-0008/#programming-recommendations
it doesn't affect much here, but there might be times in which `{}`, `""`, `[]`, `()` are valid inputs, but they are evaluated to false, hence it's a better practice to compare to `is not None`